subsys/bluetooth/host/hci_core.c

```diff
@@ -834,6 +873,15 @@ static void le_remote_feat_complete(struct net_buf *buf)
 		       sizeof(conn->le.features));
 	}
 
+	if (BT_FEAT_LE_DLE(bt_dev.le.features)) {
```

Comment:
Don't you also want to check if the remote supports the feature, in addition to the local controller supporting it?